massgov/mayflower-patternlab
# DP-7907: [Search] Adds support to callout link molecule for details variant

The search application requires, based on the recent news item results designs, a variant of the Callout Link molecule that includes an eyebrow, time details, publishing source and white background color variant.

## Description
Adds 'details' support for the Callout Link Molecule in Mayflower and a white theme variant. The aim of adding details support to the callout link molecule is to be able to meet the requirements of the search application for [recent news items](https://projects.invisionapp.com/d/#/console/13018647/274463413/preview).
<img width="312" alt="screen shot 2018-02-13 at 9 24 01 pm" src="https://user-images.githubusercontent.com/9359261/36184999-fe99090a-1104-11e8-8ae6-be1ad6bfc361.png">

## Related Issue / Ticket
- [DP-7907](https://jira.state.ma.us/browse/DP-7907)

## Steps to Test
1. Compile Mayflower branch and view in browser
2. Check template at Molecules > Callout Link with Details
3. Confirm that the variant Callout Link molecule displays with an eyebrow, time since published and source organization as shown in the screenshot below. Further, confirm that the layout is consistent on a mobile screen, as in the screenshot shown below. Confirm that the readme links to the Callout Link molecule and provides information on how to create this variant
4. Check template at Molecules > Callout Link
5. Confirm that the json variable includes eyebrow, theme, time, and source as optional. Confirm that that twig template includes conditional rendering of all of these new properties. Confirm that the readme mentions the new details variant. 
6. Check template at Molecules > Callout Link as White
7. Confirm that this variant of the Callout Link has a with background.

## Screenshots
<img width="306" alt="screen shot 2018-02-13 at 9 23 38 pm" src="https://user-images.githubusercontent.com/9359261/36185011-0dc302c8-1105-11e8-9eda-b96ea30ab653.png">

## Additional Notes:

Confirm that this does not break the already used functionality of Callout Link vanilla in higher order components (organisms and templates).
